.github/workflows/linux-nix-check.yml

```diff
@@ -0,0 +1,24 @@
+name: Build and Test on Linux Platforms
+
+on:
+  workflow_call:
+
+jobs:
+  build-and-test:
+    name: "Build and test Linux"
+    runs-on: [self-hosted, Linux, X64, aws_autoscaling]
+    steps:
+      # https://github.com/actions/checkout/issues/1552
+      - name: Clean up after previous checkout
+        run: chmod +w -R ${GITHUB_WORKSPACE}; rm -rf ${GITHUB_WORKSPACE}/*;
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Run checks
+        run: nix build -L .?#checks.x86_64-linux.all-gcc
```

Comment:
why only build without check?

Reply:
There are two chains to build now - gcc and clang.
It would take too long to build all. I would prefer to start 2 different jobs, so the CI will run faster